emulator/blockchain.go

```diff
@@ -506,23 +517,20 @@ func configureFVM(blockchain *Blockchain, conf config, blocks *blocks) (*fvm.Vir
 	cadenceLogger := conf.Logger.Hook(CadenceHook{MainLogger: &conf.ServerLogger}).Level(zerolog.DebugLevel)
 
 	config := runtime.Config{
-		Debugger:                 blockchain.debugger,
-		AccountLinkingEnabled:    true,
-		AttachmentsEnabled:       true,
-		CoverageReportingEnabled: conf.CoverageReportingEnabled,
-	}
-	coverageReportedRuntime := &CoverageReportedRuntime{
-		Runtime:        runtime.NewInterpreterRuntime(config),
-		CoverageReport: runtime.NewCoverageReport(),
-		Environment:    runtime.NewBaseInterpreterEnvironment(config),
-	}
-	customRuntimePool := reusableRuntime.NewCustomReusableCadenceRuntimePool(
-		1,
-		config,
-		func(config runtime.Config) runtime.Runtime {
-			return coverageReportedRuntime
-		},
-	)
+		Debugger:              blockchain.debugger,
+		AccountLinkingEnabled: true,
+		AttachmentsEnabled:    true,
+	}
+	// When Emulator is used as a library, a CoverageReport can be injected
+	// by calling `WithCoverageReport`, and we want to be using that one.
+	// When Emulator is used from within flow-cli, we just create a new one.
+	if conf.CoverageReport != nil {
+		config.CoverageReport = conf.CoverageReport
+		blockchain.coverageReport = conf.CoverageReport
+	} else if conf.CoverageReportingEnabled {
+		config.CoverageReport = runtime.NewCoverageReport()
+		blockchain.coverageReport = config.CoverageReport
+	}
```

Comment:
maybe better to move this to default config value ?

Reply:
But I thought that we didn't want to always enable coverage reporting. Can you elaborate on the desired change?

Reply:
instead of `WithCoverageReportingEnabled`, `WithCoverageReport(newCoverageReport())`. Now putting `WithCoverageReport(newReport)` and `WithCoverageReportingEnabled(false)` doesn't make sense. 

If we remove `WithCoverageReportingEnabled` and if coverage flag is set, we can just use ``WithCoverageReport(newCoverageReport())`.

Reply:
Could we maybe make these changes (if any) in a follow-up PR? 

For now we would just like to release the updated Emulator, so we can deploy Cadence v0.39 in the upcoming spork (flow-go's integration tests use the Emulator).

Reply:
@bluesign Could you please open a new issue and we can discuss the coverage-related configuration changes there? For now, I'll merge what we have at the moment

Reply:
@bluesign The `flow-cli` enables coverage reporting with something like:

```bash
flow emulator --coverage-reporting
```

So it only uses the `WithCoverageReportingEnabled`, it does not have to provide a `CoverageReport`. But certain libraries that use `flow-emulator`, may want to pass their own `CoverageReport` instead, that's why I have both versions.